apps/src/templates/teacherNavigation/PageHeader.tsx

```diff
@@ -28,7 +34,24 @@ const PageHeader: React.FC = () => {
   const isLoadingSectionData = useAppSelector(
     state => state.teacherSections.isLoadingSectionData
   );
+  const [ageGatedModalOpen, setAgeGatedModalOpen] = useState(false);
+  const toggleAgeGatedModal = () => {
+    setAgeGatedModalOpen(!ageGatedModalOpen);
+  };
```

Comment:
You might want to wrap this with `useCallback`?

Reply:
Interesting, so the purpose is to prevent accidental re-renders because React thinks the callback function changed.